### PR TITLE
Java11 native provider

### PR DESCRIPTION
**Pull request by kittylyst**

### CODE REVIEW

**Comment by jeffalder** [comment]
@kittylyst I love this, Thanks! It's great to have an updated http client. It'll simplify the JFR reporter, for sure.

There are no updates to telemetry_examples; that would be useful given the new `provider` method.

What lead you to make `provider` static? I wouldn't expect a Sender to be created multiple times; I think it should generally be created once and reused.

It looks like the PR build failed; are you able to see the results? If so, can you fix up the PR to fix the errors?

**Comment by kittylyst** [comment]
Thanks Jeff. The issue with the PR is that the integration_test target is broken out of the box on my machine, and rather than trying to fix that (not a Docker) expert I decided to get the refactoring done first. :)

I need to figure out why my local integration tests are broken - but that's another story. For now, I've simply configured an OkHttp provider to use with the tests - but we should definitely test this with the JDK11 provider as well.

provider() is static because that's how the builder() methods are. The original code created a new OkHttp provider on each build. I agree with you - these things should be reused, so making it static with an explicit configuration step - that's why that was part of the refactor that moved the SimpleBatch*Sender classes out of the OkHttp provider.

File: telemetry_core/src/main/java/com/newrelic/telemetry/metrics/SimpleMetricBatchSender.java [hunk 1]
```diff
@@ -17,6 +17,12 @@
  */
 public class SimpleMetricBatchSender {
 
+  private static Function<Duration, HttpPoster> ctor = null;
```
Comment (jkwatson) [inline telemetry_core/src/main/java/com/newrelic/telemetry/metrics/SimpleMetricBatchSender.java]: I really don't like adding static state here. I'd much prefer, if we want this class to support something other than okhttp, that we do that as a separate step. Can you remove this part from the PR, and we'll figure out a better way to do this as a follow-on?

File: telemetry_core/src/main/java/com/newrelic/telemetry/spans/SimpleSpanBatchSender.java [hunk 2]
```diff
@@ -18,6 +17,12 @@
  */
 public class SimpleSpanBatchSender {
 
+  private static Function<Duration, HttpPoster> ctor = null;
```
Comment (jkwatson) [inline telemetry_core/src/main/java/com/newrelic/telemetry/spans/SimpleSpanBatchSender.java]: same comment as above.

File: integration_test/src/test/java/com/newrelic/telemetry/MetricApiIntegrationTest.java [hunk 3]
```diff
@@ -18,11 +18,7 @@
 import com.newrelic.telemetry.exceptions.RetryWithBackoffException;
 import com.newrelic.telemetry.exceptions.RetryWithRequestedWaitException;
 import com.newrelic.telemetry.exceptions.RetryWithSplitException;
-import com.newrelic.telemetry.metrics.Count;
-import com.newrelic.telemetry.metrics.Gauge;
-import com.newrelic.telemetry.metrics.MetricBatchSender;
-import com.newrelic.telemetry.metrics.MetricBuffer;
-import com.newrelic.telemetry.metrics.Summary;
+import com.newrelic.telemetry.metrics.*;
```
Comment (jkwatson) [inline integration_test/src/test/java/com/newrelic/telemetry/MetricApiIntegrationTest.java]: much as it pains me to say it, I think our coding standards don't allow wildcard imports.

File: telemetry-http-java11/src/main/java/com/newrelic/telemetry/Java11HttpPoster.java [hunk 4]
```diff
@@ -0,0 +1,61 @@
...
+          httpClient.send(
+              req, java.net.http.HttpResponse.BodyHandlers.ofString(Charset.defaultCharset()));
+
+      return of(response);
+    } catch (URISyntaxException | InterruptedException e) {
+      throw new IOException(e);
+    }
+  }
+
+  public static HttpResponse of(java.net.http.HttpResponse actual) {
```
Comment (jkwatson) [inline telemetry-http-java11/src/main/java/com/newrelic/telemetry/Java11HttpPoster.java]: does this need to be public?

File: integration_test/src/test/java/com/newrelic/telemetry/MetricApiIntegrationTest.java (hunk 3 above)
Comment (jkwatson) [reply integration_test/src/test/java/com/newrelic/telemetry/MetricApiIntegrationTest.java]: ref here: https://google.github.io/styleguide/javaguide.html#s3.3.1-wildcard-imports

File: telemetry-http-java11/src/main/java/com/newrelic/telemetry/Java11HttpPoster.java (hunk 4 above)
Comment (kittylyst) [reply telemetry-http-java11/src/main/java/com/newrelic/telemetry/Java11HttpPoster.java]: I was in 2 minds about that - it makes it a trifle easier to mock / test but I'm not that fussed either way - your call.

Comment (jkwatson) [reply telemetry-http-java11/src/main/java/com/newrelic/telemetry/Java11HttpPoster.java]: If we want it public, I think it needs a better name than `of`.  How about `convertToSdkResponse`?

Comment (kittylyst) [reply telemetry-http-java11/src/main/java/com/newrelic/telemetry/Java11HttpPoster.java]: I'd prefer toSdkResponse() as it conserves the loose grammatical rule that prepositions should be preferred for the names of these types of converters.

File: telemetry-http-okhttp/README.md [hunk 5]
```diff
@@ -2,9 +2,6 @@ This module contains an implementation of the required http client interface for
 
 * `OkHttpPoster` : Implements the `HttpPoster` interface, using okhttp.
 
-In addition, it has two builder shims for creating telemetry senders that use okhttp: 
-`com.newrelic.telemetry.SimpleMetricBatchSender` and `com.newrelic.telemetry.SimpleSpanBatchSender`
-
```
Comment (jkwatson) [inline telemetry-http-okhttp/README.md]: this comment should probably still be here until we can revisit the `Simple*` stuff

**Pull request merged**